sklearn/model_selection/_search.py

```diff
@@ -360,9 +367,30 @@ def __repr__(self):
             self.parameters)
 
 
-class BaseSearchCV(six.with_metaclass(ABCMeta, BaseEstimator,
-                                      MetaEstimatorMixin)):
-    """Base class for hyper parameter search with cross-validation."""
+class _ScoreTuple (namedtuple('_ScoreTuple',
+                              ('parameters',
+                               'score'))):
+    # A raw namedtuple is very memory efficient as it packs the attributes
+    # in a struct to get rid of the __dict__ of attributes in particular it
+    # does not copy the string for the keys on each instance.
+    # By deriving a namedtuple class just to introduce the __repr__ method we
+    # would also reintroduce the __dict__ on the instance. By telling the
+    # Python interpreter that this subclass uses static __slots__ instead of
+    # dynamic attributes. Furthermore we don't need any additional slot in the
+    # subclass so we set __slots__ to the empty tuple.
+    __slots__ = ()
+
+    def __repr__(self):
+        """Simple custom repr to summarize the main info"""
+        return "score: {0:.5f}, params: {1}".format(
+            self.score,
+            self.parameters)
```

Comment:
@amueller This looks like a good way to store the `score` and `parameters`... WDYT?